pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -122,6 +123,33 @@ class CreateVM extends React.Component {
             break;
         case 'source':
             this.setState({ [key]: value });
+
+            // Clears the previously set timer.
+            clearTimeout(this.typingTimeout);
+
+            if (this.state.sourceType == URL_SOURCE && this.state.vendor == NOT_SPECIFIED) {
+                const autodetectOSHandler = (os) => autodetectOS(os)
+                        .then(res => {
+                            const osName = res
+                                    .split('for OS')
+                                    .pop()
+                                    .replace(/['"\n]+/g, '')
+                                    .trim();
```

Comment:
This is a bit awkward. Not sure if the "env" format is helpful in any way:
```
❱❱❱ osinfo-detect --format=env https://ftp.fau.de/fedora/linux/releases/29/Server/x86_64/os/images/boot.iso
OSINFO_BOOTABLE=1
OSINFO_INSTALLER=http://fedoraproject.org/fedora/29
OSINFO_MEDIA=http://fedoraproject.org/fedora/29:3

❱❱❱ osinfo-detect --format=env --type=tree https://ftp.fau.de/fedora/linux/releases/29/Server/x86_64/os/
OSINFO_INSTALLER=http://fedoraproject.org/fedora/29
OSINFO_TREE=http://fedoraproject.org/fedora/29:1
OSINFO_TREE_KERNEL=images/pxeboot/vmlinuz
OSINFO_TREE_INITRD=images/pxeboot/initrd.img
OSINFO_TREE_BOOT_ISO=images/boot.iso
```
These don't nicely map to the strings that we have in the selectors, but maybe these strings have corresponding IDs which match that output? I hope these "env" names don't come out of thin air. :-)

If the "env" format is unsuitable, do you mind changing that to a regular expression? That's a little easier to grasp and more robust against format changes:
```
> s = "Media is bootable.\nMedia is an installer for OS 'Fedora 29 Server'"
> s.match(/OS ['"](.*)['"]/)[1]
'Fedora 29 Server'
```

Reply:
I was already disappointed with this. The env vars don't seem to help at all basically. I am ok with your regexp, I will adjust my code! Thanks :)